control-plane/src/health.rs

```diff
@@ -73,32 +85,35 @@ async fn run_ecs_health_check_service(
         .map_err(ServerError::from)
 }
 
-macro_rules! unwrap_or_err {
-    ($result:expr) => {
-        match $result {
-            Ok(ok) => ok,
-            Err(error) => {
-                return HealthCheckLog::new(HealthCheckStatus::Err, Some(error.to_string()))
-            }
-        }
-    };
-}
+async fn health_check_data_plane() -> Result<HealthCheckVersion, ServerError> {
+    let stream = get_connection_to_enclave(ENCLAVE_HEALTH_CHECK_PORT).await?;
+
+    let (mut sender, connection) = hyper::client::conn::handshake(stream).await?;
 
-async fn health_check_data_plane() -> HealthCheckLog {
-    let stream = unwrap_or_err!(get_connection_to_enclave(ENCLAVE_HEALTH_CHECK_PORT).await);
-    let (mut sender, connection) = unwrap_or_err!(hyper::client::conn::handshake(stream).await);
     tokio::spawn(connection);
     let request = Request::builder()
         .method("GET")
         .header("User-Agent", "CageHealthChecker/0.0")
         .body(Body::empty())
         .expect("Cannot fail");
-    let response = unwrap_or_err!(sender.send_request(request).await);
-    unwrap_or_err!(serde_json::from_slice(
-        &hyper::body::to_bytes(response.into_parts().1)
-            .await
-            .unwrap()[..]
-    ))
+
+    let response = sender.send_request(request).await?;
+    let (parts, response) = response.into_parts();
+
+    let content_type = parts
+        .headers
+        .get("Content-Type")
+        .map(HeaderValue::to_str)
+        .and_then(Result::ok);
+
+    let bytes = &hyper::body::to_bytes(response).await?;
+
+    Ok(match content_type {
+        Some("application/json;version=1") => {
+            HealthCheckVersion::V1(serde_json::from_slice::<HealthCheckLog>(bytes)?)
+        }
+        _ => HealthCheckVersion::V0(serde_json::from_slice::<HealthCheckLog>(bytes)?),
+    })
```

Comment:
Doing the match in the `Ok` is pretty messy, cleaner to split out:
```suggestion
  let healthcheck_result = match content_type {
        Some("application/json;version=1") => {
            HealthCheckVersion::V1(serde_json::from_slice::<HealthCheckLog>(bytes)?)
        }
        _ => HealthCheckVersion::V0(serde_json::from_slice::<HealthCheckLog>(bytes)?),
    };
    Ok(healthcheck_result)
```

Also might be more robust to include the mime crate to parse the header and it's attributes.

Reply:
I think this is just an unnecessary alloc?

wrt to mime crate I'm to leave as is in this PR but can investigate improvements later

Reply:
would assume it's compiled away

Reply:
excellent point